### ISSUE TITLE
False positive `nullable type must be written as "?(A&B)"`

### ISSUE THREAD

**Issue by tminich** [hammer-tools/php-hammer#28]
I have set the inspection to prefer short syntax. Unfortunately it doesn't properly detect DNF-types:

```php
interface A {}
interface B {}

function test((A&B)|null $param = null): void {}
```
triggers the warning, but
```php
function test(?(A&B) $param = null): void {}
```
is not valid php.

**Comment by rentalhost** [comment]
Nice catch! Thanks!

**Comment by rentalhost** [comment]
Thank you for taking the time to report this issue. The fix has been implemented in version **0.24.0** and is available on the **[Releases](https://github.com/hammer-tools/php-hammer/releases)** page for manual installation. Alternatively, you can wait for the official release on the **JetBrains Marketplace**, which should happen in approximately 3 days.

- **Regarding the issue you reported:** Now, the scenario you mentioned should no longer pose any problems (i.e., it will accept `(A&B)|null` even if you choose to use the _shortened format_, as PHP doesn't support `?` before intersected types). I've also updated other inspections that deal with similar issues to avoid problems.
